tests/diag.sh

```diff
@@ -284,11 +298,13 @@ case $1 in
 		   echo Shutting down instance 2
 		fi
 		. $srcdir/diag.sh wait-queueempty $2
+		cp rsyslog$2.pid rsyslog$2.pid.save
```

Comment:
So if something else can remove the pid file, won't this potentially fail?

I think you have to check success/failure of `cp` here and then perform the kill conditionally off the saved pid file, no?

The read of the pid file needs to be atomic, something like:

```
rsyslog_pid="$(cat rsyslog$2.pid 2> /dev/null)"
if [ "x$rsyslog_pid" != "x" ]; then
    ./msleep 1000
    kill $rsyslog_pid
fi
```